c7n/resources/vpc.py

```diff
@@ -241,6 +241,80 @@ def process(self, resources, event=None):
         return results
 
 
+@Vpc.filter_registry.register('dhcp-options')
+class DhcpOptionsFilter(Filter):
+    """Filter VPCs based on their dhcp options
+
+    """
+    schema = type_schema(
+        'dhcp-options',
+        domainname={'type': 'array', 'items': {'type': 'string'}},
+        nameserver={'type': 'array', 'items': {'type': 'string'}},
+        ntpserver={'type': 'array', 'items': {'type': 'string'}})
+    permissions = ('ec2:DescribeDhcpOptions',)
+
+    def validate(self):
```

Comment:
this entire validate method could be replaced by a better jsonschema definition.